h2o-core/src/main/java/hex/grid/filter/PermutationFilter.java

```diff
@@ -8,18 +8,13 @@
  * @param <MP> as {@link hex.grid.HyperSpaceWalker} is working only with descendants of the {@link Model.Parameters},
  *            this interface is generic with corresponding type parameter
  */
-public interface PermutationFilter<MP extends Model.Parameters> {
+public interface PermutationFilter<MP extends Model.Parameters> extends Resetable {
 
   /**
    * Main method that decides whether given {@code permutation} should be skipped
    * @param permutation instance of {@link Model.Parameters} that represents particular
    *                   combination of hyper parameters during grid search
-   * @return `true` when permutation should be skipped
+   * @return `false` when permutation should be skipped
    */
-  boolean permutationIsSkipped(MP permutation);
-
-  /**
-   * As filtering logic might be stateful we need an ability to reset it to an initial state
-   */
-  void reset();
+  boolean test(MP permutation);
```

Comment:
looks like a resetable predicate now ;)

Reply:
@seb-h2o and it is. I don't want to use Predicate + Resetable as it looks better to have alias `PermutationFilter` when you define filters:

```
PermutationFilter<GBMModel.GBMParameters> defaultPermutationFilter = new AnyMatchPermutationFilter(strictFilterFunction);
```